packages/desktop/src/services/FileDownloadServiceElectron.ts

```diff
@@ -352,4 +534,32 @@ export default class FileDownloadServiceElectron
             });
         });
     }
+
+    private parseS3Url(url: string): { bucket: string; key: string; region: string } {
+        const { hostname, pathname } = new URL(url);
+        const [bucket] = hostname.split(".");
+        const key = pathname.slice(1);
+        const region = hostname.split(".")[2];
+        return { bucket, key, region };
+    }
+
+    private async listS3Objects(bucket: string, prefix: string, region: string): Promise<string[]> {
```

Comment:
I take it zarrita can't do this for us?

Reply:
There is some merit to the flatmap function of zarrita for listing contents but in dealing with it you have to make a lot of assumptions (I think) so it didn't end up working too well for me.

Reply:
🆗